src/domain.rs

```diff
@@ -548,13 +558,31 @@ impl<F> Domain<F> {
         }
     }
 
-    #[cfg(not(loom))]
-    fn calc_shard(input: *mut Retired) -> usize {
+    #[cfg(all(not(loom), feature = "multiply_hash"))]
+    fn calc_shard(&self, input: *mut Retired) -> usize {
+        //XXX: This is not prime if a usize 32 bits
+        const LARGE_PRIME: usize = 4445950232728569541;
```

Comment:
I wonder if you can use a `#[cfg]` on [`target_pointer_width`](https://doc.rust-lang.org/nightly/reference/conditional-compilation.html#target_pointer_width) here to use a different prime for `u32`-sized pointers?